src/dct/ai/Commander.lua

```diff
@@ -206,6 +205,42 @@ function Commander:getMission(id)
 	return self.missions[id]
 end
 
+--[[
+-- return the number of missions that can be assigned per given type
+--]]
+function Commander:getAvailableMissions(missionTypes)
+	local assetmgr = self.theater:getAssetMgr()
+
+	-- map asset types to the given mission type names
+	local assetTypeMap = {}
+	for missionTypeName, missionTypeId in pairs(missionTypes) do
+		for assetType, _ in pairs(enum.missionTypeMap[missionTypeId]) do
+			assetTypeMap[assetType] = missionTypeName
+		end
+	end
+
+	local tgts = assetmgr:getTargets(self.owner, assetTypeMap)
```

Comment:
The whole goal was to remove target identification from the asset manager and let commanders manage target lists. In the next branch there is a patch that does just this.

Reply:
Alright, now it's on the `next` branch, and uses the commander's own target list. Thanks for the heads up.